### PR TITLE
Add Time Series Classification for Blood Glucose Message Type Prediction

### PR DESCRIPTION
**Pull request by andytubeee**
This PR introduces a time-series classification model to predict message types (msg_type) from blood glucose levels (bgl) using the sktime library. The implementation leverages the TimeSeriesForestClassifier and includes preprocessing steps such as label encoding and handling missing data. Key features of the update include:

- Loading and merging multiple CSV files from the data/raw directory.
- Preprocessing the data by filling missing values and encoding the msg_type target variable.
- Splitting the data into training and test sets using temporal_train_test_split for time-series compatibility.
- Training a TimeSeriesForestClassifier on the bgl feature to predict the msg_type.
- Evaluating the model using accuracy and weighted F1 score.

Key Changes:

- Added data loading from multiple CSV files in data/raw/.
- Implemented label encoding for msg_type.
- Trained a time-series classifier on the bgl feature.
- Evaluated performance metrics such as accuracy and weighted F1 score.

Next Steps:

- Explore additional features (e.g., affects_fob, affects_iob) to improve model performance.
- Experiment with alternative time-series models for better accuracy.

### CODE REVIEW

**Comment by andytubeee** [comment]
**Note:** Result is not promising as the model only returns NULL due to imbalanced msg_type from training

**Comment by RobotPsychologist** [comment]
Thanks for this @andytubeee 

We need to structure this as a time series annotation problem. I do not expect that we would be able to classify msg_types from bgl because it's usually the msg_type that indicates a causal effect on bgl in most cases. Look up what change points are and see what I'm in the middle of building in notebook 1.01

Please have a read through the section on naming conventions for our project:
https://cookiecutter-data-science.drivendata.org/using-the-template/

Can we separate the data generation notebook from the modeling notebook? We're going to have so many people working on this, so I need to start with strong adherence to this standard. People need to know which notebooks to look through if they want to reuse something you've created. Most of the reused code will be from the data generation notebooks.

**Comment by andytubeee** [comment]
Meal identification is using bgl data to predict if a meal intake has occurred right? What other indicators can we use other than msg_type?

**Comment by andytubeee** [comment]
I see now, this is an unsupervised problem

**Comment by RobotPsychologist** [comment]
Right now I'm building the change points data to occur when there is a meal announcement, and then 2,3,4,or 5 hours post meal announcement.  

It's semi-supervised, but were starting out with the supervised approach and will need to build methods for assessing it in a non-supervised setting because some of the meals are labelled with endpoints and some meals won't be.
